lib/line-index/src/tests.rs

```diff
@@ -0,0 +1,73 @@
+use super::*;
+
+#[test]
+fn test_line_index() {
```

Comment:
Let's move this to an integration test, for libs we do want to drive the tests trough the public API, to ensure that everything is public that needs to be.

Reply:
i'm not sure what you're saying here. should i move the tests to where the one slow test is, here? https://github.com/rust-lang/rust-analyzer/blob/ffdbfe73384f88f666a1fbb762407afc90b48280/crates/ide-db/src/tests/line_index.rs

> for libs we do want to drive the tests trough the public API, to ensure that everything is public that needs to be.

are the existing tests not sufficient for this? I can see about adding more.

Reply:
I mean `mv lib/line-index/src/tests.rs lib/line-index/tests/it.rs`, so that the tests are in differenc CU. We don't want that for internal crates, but we do want that for crates with public APIs.

Reply:
ok i moved all tests except one which asserts about the internals